app/controllers/authentication_controller.rb

```diff
@@ -9,7 +9,17 @@ def self.call(env)
 
   def redirect_to_omniauth
     store_location(attempted_path) if request.get?
-    redirect_to "/auth/#{default_provider}"
+    if default_provider == "auth0" && e2e_user?
+      # the value passed in connection must match the name given in the auth0 terraform
+      redirect_to "/auth/#{default_provider}?connection=Username-Password-Authentication"
+    else
+      redirect_to "/auth/#{default_provider}"
+    end
+  end
+
+  def e2e_user?
+    # this is appended to the first request in the e2e tests to activate the username/password flow
+    request.params.key? "force_credentials_flow"
```

Comment:
I think maybe we should go with a more general parameter name/value combination, to give us more flexibility in future. Something like `?auth=e2e`?

```suggestion
    request.params[:auth] == "e2e"
```

Reply:
I've made this change now, thanks for the suggestion!